include/props.h

```diff
@@ -54,123 +77,1286 @@ typedef struct plist *PropPtr;
+#define PropDataStr(x) ((x)->data.str)      /* String         */
+#define PropDataVal(x) ((x)->data.val)      /* Integer        */
+#define PropDataRef(x) ((x)->data.ref)      /* DBREF          */
+#define PropDataLok(x) ((x)->data.lok)      /* Lock           */
+#define PropDataFVal(x) ((x)->data.fval)    /* Floating Point */
+
+/* Get prop name */
 #define PropName(x) ((x)->key)
 
+/* Set and get property flags */
 #define SetPFlags(x,y) {(x)->flags = ((x)->flags & PROP_TYPMASK) | (unsigned short)y;}
 #define PropFlags(x) ((x)->flags & ~PROP_TYPMASK)
 
+/* Set and get property type */
 #define SetPType(x,y) {(x)->flags = ((x)->flags & ~PROP_TYPMASK) | (unsigned short)y;}
 #define PropType(x) ((x)->flags & PROP_TYPMASK)
 
+/* Set and get the raw flags (unsigned short integer) */
 #define SetPFlagsRaw(x,y) {(x)->flags = (unsigned short)y;}
 #define PropFlagsRaw(x) ((x)->flags)
 
+/* Check if property is blessed */
 #define Prop_Blessed(obj,propname) (get_property_flags(obj, propname) & PROP_BLESSED)
 
-/* property access macros */
+/* property access macros
+ *
+ * If you change any of these, your MUCK will be drastically incompatible
+ * with pretty much everything.
+ */
 
 #define PROP_RDONLY '_'
 #define PROP_RDONLY2 '%'
 #define PROP_PRIVATE '.'
 #define PROP_HIDDEN '@'
 #define PROP_SEEONLY '~'
 
+/* Macros to check property access types */
 #define Prop_ReadOnly(name) \
     (Prop_Check(name, PROP_RDONLY) || Prop_Check(name, PROP_RDONLY2))
 #define Prop_Private(name) Prop_Check(name, PROP_PRIVATE)
 #define Prop_SeeOnly(name) Prop_Check(name, PROP_SEEONLY)
 #define Prop_Hidden(name) Prop_Check(name, PROP_HIDDEN)
 #define Prop_System(name) is_prop_prefix(name, "@__sys__")
 
+/**
+ * Check each segment of the property path and see if 'what' is the first
+ * character of the segment.
+ *
+ * It is recommended you use the property check macro's instead:
+ * Prop_ReadOnly, Prop_Private, Prop_SeeOnly, Prop_Hidden, Prop_System
+ *
+ * @param name Property name string
+ * @param what Single character
+ * @return true if 'what' appears in any path segment.
+ */
 int Prop_Check(const char *name, const char what);
-void add_prop_nofetch(dbref player, const char *type, const char *strval, int value);
-void add_property(dbref player, const char *type, const char *strval, int value);
+
+/**
+ * Adds a new property to an object without diskbase handling.  It is
+ * safer to use the add_property method, however this is exposed because
+ * it used in both db.c and property.c
+ *
+ * This is a wrapper around set_property_nofetch, so if you need to
+ * set values more complex than string or integer, use set_property_nofetch
+ * instead.
+ *
+ * @internal
+ * @param player Object to set property on
+ * @param pname Property name
+ * @param strval String value - pass NULL if you want an integer value
+ * @param value Integer value - ignored if String value is used.
+ */
+void add_prop_nofetch(dbref player, const char *pname, const char *strval,
```

Comment:
Change from type to pname intentional?

Reply:
Yeah.  Whomever made the header file did a lot of "copy pasta" and 'type' was used in place of name almost everywhere.  pname is correct in all cases and matches the definition of the function which has the correct name.